<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AskMultisigDialog</class>
 <widget class="QDialog" name="AskMultisigDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>566</width>
    <height>475</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>550</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Multisig</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="sizeConstraint">
    <enum>QLayout::SetMinimumSize</enum>
   </property>
   <item><widget class="QLineEdit" name="labelEdit"/></item>
   <item>
    <widget class="QToolBox" name="toolBox">
     <property name="frameShape">
      <enum>QFrame::Panel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="pageEscrow">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>546</width>
        <height>326</height>
       </rect>
      </property>
      <attribute name="label">
       <string>Create standard Escrow multisig</string>
      </attribute>
      <layout class="QGridLayout" name="escrowGridLayout">
       <item row="4" column="2">
        <widget class="QLineEdit" name="counterpartyAddressLineEdit"/>
       </item>
       <item row="5" column="2">
        <widget class="QLineEdit" name="escrowAddressLineEdit"/>
       </item>
       <item row="1" column="2">
        <widget class="QLineEdit" name="yourAddressLineEdit">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="font">
          <font>
           <italic>true</italic>
          </font>
         </property>
         <property name="text">
          <string>(will be generated automatically)</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="yourAddressLabel">
         <property name="text">
          <string>Your address pubkey</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QLabel" name="escrowLabel">
         <property name="text">
          <string>Escrow pubkey &lt;i&gt;(optional)&lt;/i&gt;&amp;nbsp;</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QLabel" name="counterpartyAddressLabel">
         <property name="text">
          <string>Counterparty pubkey</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1" colspan="2">
        <widget class="QLabel" name="escrowExplainedLabel">
         <property name="styleSheet">
          <string notr="true">background-color: #f0fbf4</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::Panel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="text">
          <string>&lt;p&gt;&lt;i&gt;Your address pubkey&lt;/i&gt; is generated automatically.
You only need to provide a address pubkey generated by a person you are trading with (&lt;i&gt;Counterparty pubkey&lt;/i&gt;) and optionally third party (the escrow) one.&lt;/p&gt;
&lt;p&gt;If no escrow pubkey is provided a 2/2 multisig would be generated
(i.e. both you and the counterparty need to agree to redeem). If escrow is provided, 2/3 multisig would be generated (i.e. two of you three need to agree).&lt;/p&gt;
&lt;p&gt;To generate raw pubkey click button below. To import multisig address generated by a counterparty, switch to last tab.&lt;/p&gt;</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignJustify|Qt::AlignTop</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
         <property name="margin">
          <number>4</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pageCustom">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>546</width>
        <height>326</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <attribute name="label">
       <string>Create custom multisig</string>
      </attribute>
      <layout class="QFormLayout" name="formLayout">
       <property name="sizeConstraint">
        <enum>QLayout::SetMinimumSize</enum>
       </property>
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="signaturesRequiredLabel">
         <property name="text">
          <string>Signatures required</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="signaturesRequiredSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>20</number>
         </property>
         <property name="value">
          <number>3</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="totalAdressesLabel">
         <property name="text">
          <string>Total number of pubkeys</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="totalAdressesSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>20</number>
         </property>
         <property name="value">
          <number>2</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="foreignAdressesLabel">
         <property name="maximumSize">
          <size>
           <width>135</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>&lt;p&gt;Foreign pubkeys&lt;/p&gt;&lt;p style=&quot; font-style:italic;color:#555&quot;&gt;(remaining pubkeys will be generated and added to your wallet)&lt;/p&gt;</string>
         </property>
         <property name="textFormat">
          <enum>Qt::RichText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPlainTextEdit" name="foreignAdressesEdit"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pageImport">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>546</width>
        <height>326</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <attribute name="label">
       <string>Import multisig created by your counterparty</string>
      </attribute>
      <layout class="QVBoxLayout" name="importVerticalLayout">
       <item>
        <widget class="QLabel" name="importLabel">
         <property name="font">
          <font>
           <italic>true</italic>
          </font>
         </property>
         <property name="text">
          <string>Enter multisig redeem script below:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="importRedeemScriptEdit">
         <property name="font">
          <font>
           <family>Lucida Console</family>
          </font>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>AskMultisigDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>AskMultisigDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
